src/tests/docker.md

```diff
@@ -1,36 +1,44 @@
+helpful to test environments different from your local system. You will
 need to install Docker on a Linux, Windows, or macOS system (typically Linux
 will be much faster than Windows or macOS because the latter use virtual
-machines to emulate a Linux environment). To enter interactive mode which will
-start a bash shell in the container, run `src/ci/docker/run.sh --dev <IMAGE>`
-where `<IMAGE>` is one of the directory names in `src/ci/docker` (for example
-`x86_64-gnu` is a fairly standard Ubuntu environment).
-
-The docker script will mount your local Rust source tree in read-only mode,
-and an `obj` directory in read-write mode. All of the compiler artifacts will
-be stored in the `obj` directory. The shell will start out in the `obj`
-directory. From there, you can run `../src/ci/run.sh` which will run the build
-as defined by the image.
-
-Alternatively, you can run individual commands to do specific tasks. For
-example, you can run `../x test tests/ui` to just run UI tests.
-Note that there is some configuration in the [`src/ci/run.sh`] script that you
-may need to recreate. Particularly, set `submodules = false` in your
-`config.toml` so that it doesn't attempt to modify the read-only directory.
+machines to emulate a Linux environment).
+
+Jobs running in CI are configured through a set of bash scripts, and it is not always trivial to reproduce their behavior locally. If you want to run a CI job locally in the simplest way possible, you can use a provided helper Python script that tries to replicate what happens on CI as closely as possible:
+
+```bash
+python3 src/ci/github-actions/ci.py run-local <job-name>
+# For example:
+python3 src/ci/github-actions/ci.py run-local dist-x86_64-linux-alt
+```
 
-Some additional notes about using the Docker images:
+If the above script does not work for you, you would like to have more control of the Docker image execution, or you want to understand what exactly happens during Docker job execution, then continue reading below.
 
+## The `run.sh` script
+The [`src/ci/docker/run.sh`] script is used to build a specific Docker image, run it,
+build Rust within the image, and either run tests or prepare a set of archives designed for distribution. The script will mount your local Rust source tree in read-only mode, and an `obj` directory in read-write mode. All the compiler artifacts will be stored in the `obj` directory. The shell will start out in the `obj`directory. From there, it will execute `../src/ci/run.sh` which starts the build as defined by the Docker image.
+
+You can run `src/ci/docker/run.sh <image-name>` directly. A few important notes regarding the `run.sh` script:
+- There is some configuration used on CI that you may need to recreate. In particular, set `submodules = false` in your `config.toml` so that it doesn't attempt to modify the read-only directory.
```

Comment:
If we are running a docker image, it already sets `submodules = false` if I am not mistaken.

We should instead say "make sure submodules are fetched with `git ...`" I think.

Reply:
This always confuses me, I could swear that the Docker build sometimes took data from my root config.toml :) But it seems like it doesn't happen, and yeah, the build expects submodules to be downloaded, because that's what happens on CI.

I reworded the paragraph, let me know what you think.